llvmlite/tests/test_refprune.py

```diff
@@ -478,23 +481,45 @@ def test_fanout_raise_3(self):
         self.assertEqual(stats.fanout_raise, 2)
 
     fanout_raise_4 = r"""
-define i32 @main(i8* %ptr, i1 %cond) {
+define i32 @main(i8* %ptr, i1 %cond, i8** %excinfo) {
 bb_A:
     call void @NRT_incref(i8* %ptr)
     br i1 %cond, label %bb_B, label %bb_C
 bb_B:
-    ret i32 1, !ret_is_raise !0    ; BAD; all tails are raising without decref
+    ret i32 1    ; BAD; all tails are raising without decref
 bb_C:
-    ret i32 1, !ret_is_raise !0    ; BAD; all tails are raising without decref
+    ret i32 1    ; BAD; all tails are raising without decref
 }
 
-!0 = !{i32 1}
+!0 = !{i1 1}
 """
 
     def test_fanout_raise_4(self):
         mod, stats = self.check(self.fanout_raise_4)
         self.assertEqual(stats.fanout_raise, 0)
 
+    fanout_raise_5 = r"""
+define i32 @main(i8* %ptr, i1 %cond, i8** %excinfo) {
+bb_A:
+    call void @NRT_incref(i8* %ptr)
+    br i1 %cond, label %bb_B, label %bb_C
+bb_B:
+    call void @NRT_decref(i8* %ptr)
+    br label %common.ret
+bb_C:
+    store i8* null, i8** %excinfo, !numba_exception_output !0
+    br label %common.ret
+common.ret:
+    %common.ret.op = phi i32 [ 0, %bb_B ], [ 1, %bb_C ]
+    ret i32 %common.ret.op
+}
+
```

Comment:
restore missing metadata
```suggestion
!0 = !{i1 1}
```

Reply:
https://github.com/numba/llvmlite/pull/830#discussion_r1128313228